baseapp/options.go

```diff
@@ -110,8 +111,10 @@ func (app *BaseApp) SetVersion(v string) {
 	app.version = v
 }
 
-// SetProtocolVersion sets the application's protocol version
-func (app *BaseApp) SetProtocolVersion(v uint64) {
+// SetAppVersion sets the application's protocol version
+func (app *BaseApp) SetAppVersion(ctx sdk.Context, v uint64) {
+	vp := &tmproto.VersionParams{AppVersion: v}
+	app.paramStore.Set(ctx, ParamStoreKeyVersionParams, vp)
```

Comment:
does this need to be an if statement since its modifying state?

Reply:
yeah it probably should be as this could be triggered in `InitGenesis` for v1